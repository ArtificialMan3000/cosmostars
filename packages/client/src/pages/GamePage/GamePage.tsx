import { FC, useEffect, useRef } from "react";
import { GameModalConfig, GameStatuses } from "../../shared/constants";
import { Game, initGame } from "../../entities/game/controller/Game";
import { GameModal } from "../../widgets/GameModal/GameModal";

export const GamePage: FC = () => {
  const canvasElement = useRef<HTMLCanvasElement>(null);
  const game = useRef<Game | null>(null);

  useEffect(() => {
    if (canvasElement.current) {
      game.current = initGame(canvasElement.current);
    }
  }, [canvasElement]);

  const startNewGame = () => {
    game.current?.start();
  };

  return (
    <>
      <GameModal
        {...GameModalConfig[GameStatuses.START]}
        onStart={startNewGame}
      />
      <canvas className="game-canvas" ref={canvasElement}></canvas>
    </>
  );
};
