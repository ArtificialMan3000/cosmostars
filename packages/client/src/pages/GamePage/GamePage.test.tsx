import { ReactElement } from "react";
import { BrowserRouter } from "react-router-dom";

import {
  shouldMatchSnapshot,
  shouldRenderWithoutError,
} from "../tests/factories";
import { GamePage } from "./GamePage";

let Page: ReactElement;

// ! Эти тесты падают из-за ошибки типизации. Надо поправить код.
describe.skip("Game Page", () => {
  beforeEach(() => {
    Page = (
      <BrowserRouter>
        <GamePage />
      </BrowserRouter>
    );
  });

  it("should render without error", () => {
    shouldRenderWithoutError(Page);
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(Page);
  });
});
