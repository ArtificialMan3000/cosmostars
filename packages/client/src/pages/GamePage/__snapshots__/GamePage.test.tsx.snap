// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Page should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-2lhqg7"
  >
    Score: 
    0
  </div>
  <canvas
    class="game-canvas"
  />
</div>
`;
